custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -437,10 +456,14 @@ def allowed_to_forward(self, location):
         if successful_records:
             latest_record_id = successful_records.split(" ")[-1]
             latest_record = RepeatRecord.get(latest_record_id)
-            previous_payload = latest_record.attempts[-1].payload if latest_record.attempts else {}
-            current_payload = unicode(get_bets_location_json(location))
-            if current_payload == previous_payload:
-                return False
+            previous_payload = latest_record.attempts[-1].info if latest_record.attempts else "{}"
+            try:
+                previous_payload_json = json.loads(previous_payload)
+                current_payload = get_bets_location_json(location)
+                if current_payload == previous_payload_json:
+                    return False
+            except (TypeError, ValueError):
+                pass
```

Comment:
this looks identical to https://github.com/dimagi/commcare-hq/pull/17806/commits/f3fca737a963d179b3b65616d7b5048eb3670bb4#diff-93d85770198e309ac7e7506062ffecccR421
any reason it cant be split out into a function

Reply:
They get the `current_payload` in a different way, but yes I probably could have done it better.